pkg/controllers/binding/common.go

```diff
@@ -171,6 +173,7 @@ func mergeAnnotations(workload *unstructured.Unstructured, workNamespace string,
 	annotations := make(map[string]string)
 	util.MergeAnnotation(workload, workv1alpha2.WorkNameAnnotation, names.GenerateWorkName(workload.GetKind(), workload.GetName(), workload.GetNamespace()))
 	util.MergeAnnotation(workload, workv1alpha2.WorkNamespaceAnnotation, workNamespace)
+	util.MergeAnnotation(workload, workv1alpha2.KarmadaWorkloadGenerationAnnotationKey, strconv.Itoa(int(workload.GetGeneration())))
```

Comment:
Does int64 to int conversion lose information on 32-bit systems?

Reply:
> Does int64 to int conversion lose information on 32-bit systems?

@XiShanYongYe-Chang  fixed.